Tests signup login week9

simaoalicechen opened the pull request:
Tests cover these files: 

1. All files under Signup_login;
2. modalFullScreen.vue

Tests Created: 4
Tests Passed: 4

Currently: appAccountSignup is undefined.

simaoalicechen committed:
Added tests

simaoalicechen committed:
Merge branch 'gcivil-nyu-org:develop' into develop

simaoalicechen committed:
Added tests

simaoalicechen committed:
Added pinia and plugins and commands working, only exists tests working

simaoalicechen committed:
Added a switchToSignupButton test, it is working

Currently, this cant pass the linter check, because we will have to leave some non-functioning codes in the file

simaoalicechen committed:
Added switchToLogin button, logo exists, logoImgLink exists tests, all working

I think I can keep adding static and actionable components tests from now on. The next steps is to get the API get and post call tests working. Will keep working on those tomorrow.

simaoalicechen committed:
Merge branch 'gcivil-nyu-org:develop' into develop

simaoalicechen committed:
Merge branch 'gcivil-nyu-org:develop' into Tests-Signup-Login-Week9

simaoalicechen committed:
Added signupForm sumbit.prevent test, signupForm tets, signup_components_container, and deleted API tests that do not really work.

simaoalicechen committed:
linter working

simaoalicechen committed:
Added 4 four component tests

simaoalicechen committed:
Added 2 tests to LoggedOut

simaoalicechen committed:
Added three more tests for ModalFullScreen

simaoalicechen committed:
Merge branch 'develop' into Tests-Signup-Login-Week9

simaoalicechen committed:
Merge branch 'gcivil-nyu-org:develop' into develop

simaoalicechen committed:
29 component tests in total for every file under signup_login and in the file ModalFullScreen

All these tests are passing

simaoalicechen commented:
There are 29 tests in this PR, all for files under signup_login, and for the file ModelFullScreen. They are all passing. I want to stop here for this PR, and will work on testing for other folders in frontend later, but this PR is getting too large. Please review.

simaoalicechen committed:
Merge branch 'develop' into Tests-Signup-Login-Week9

simaoalicechen commented:
I will temporarily take off the reviewers, as I intend to commit a few more things. Will add you back.

simaoalicechen committed:
Merge branch 'gcivil-nyu-org:develop' into develop

simaoalicechen committed:
Found a few bugs and added two more tests to login

simaoalicechen committed:
Added signup with google in signup. It only makes to have SSO on both signup and login

simaoalicechen committed:
Added a new test, and commented on spaces where API call should be implemented.

simaoalicechen committed:
Cleaned up the files and ready for review. 24 components tests, all passing

yalexublue commented on src/frontend/shortlist/src/components/layout/signup_login/Signup.vue:
```diff
@@ -79,8 +79,20 @@ export default {
+        class="btn btn-outline-dark"
+      >
+        Sign Up with Google
+      </button>
```
Are these changes you merged from the demo of last week?

simaoalicechen replied:
No. I found out someone left logInWithSSO but took out the signUpWithSSO, so I added it back, as I commented there: if there is no signup with SSO, how can there be login with SSO?

yalexublue commented on src/frontend/shortlist/src/components/layout/signup_login/__tests__/LoginSignup.test.js:
```diff
@@ -0,0 +1,317 @@
+    const componentWrapper = mount(Login, {
+      props: {
+        /*
+        TODO(Add API variables here)
```
login.vue does not have prop and since for unit test we do not consider axio call, I don't think we need this whole block: props: ...

simaoalicechen replied:
It does not have. But I believe it will have to have some to match API calls? I think Daniel did not know the frontend codes do not have props for login. I left them there as space for him to add API tests later.

yalexublue replied:
I think he mentioned we are not doing API call check in unit test?

simaoalicechen replied:
Nope, but you in the meeting on Friday mentioned that we somehow need to write something to mimic API testing, and then he clarified this morning that he didn't need me to do any API writing, rather just needed me to leave space and do as much component testing as possible. That's why I left the comments there so people know where to pick things up from there

yalexublue commented on src/frontend/shortlist/src/components/layout/signup_login/__tests__/LoginSignup.test.js:
```diff
@@ -0,0 +1,317 @@
+    });
+    const loginForm = componentWrapper.find("#loginForm");
+    expect(loginForm.exists()).toBe(true);
+    await loginForm.trigger("submit.prevent");
```
What exactly is this doing here, you triggered a button, but not expect any result? It's not testing anything

simaoalicechen replied:
I am not sure what you meant. I used the codes I saw from the previous test file. If you think I am missing something. Comment here which const or function I am missing? I am trying to test if this button would be trigger-red successfully.

yalexublue replied:
It's like you let the test triggered button click, you usually expect a feedback as what will happen, that's the test, but here you triggered the button, then nothing is taking place to check what happened after button press
So you valid the button exist, but didn't check what the button do

simaoalicechen replied:
Yes... Then what feedback? Can you elaborate? What did you do?

yalexublue commented on src/frontend/shortlist/src/components/layout/signup_login/__tests__/LoginSignup.test.js:
```diff
@@ -0,0 +1,317 @@
+    });
+    const loginWithGoogle = componentWrapper.find("#loginWithGoogle");
+    expect(loginWithGoogle.exists()).toBe(true);
+    await loginWithGoogle.trigger("submit.prevent");
```
Same as above, you made sure the button exist, then triggered the button, but then no expectation later, it's not testing the trigger result

simaoalicechen replied:
See above

yalexublue commented on src/frontend/shortlist/src/components/layout/signup_login/__tests__/LoginSignup.test.js:
```diff
@@ -0,0 +1,317 @@
+    });
+    const switchToSignupButton = componentWrapper.find("#switchToSignupButton");
+    expect(switchToSignupButton.exists()).toBe(true);
+    await switchToSignupButton.trigger("click");
```
Same reasoning as above

simaoalicechen replied:
See above. If you think I am missing a lot of similar functions here, this is the exact reason why I needed reviews on day one I created the first few tests

yalexublue commented on src/frontend/shortlist/src/components/layout/signup_login/__tests__/LoginSignup.test.js:
```diff
@@ -0,0 +1,317 @@
+      },
+    });
+    const bigTitles = componentWrapper.find("#big");
+    expect(bigTitles.exists()).toBe(true);
```
I don't see all these test's purpose, if you are testing all these component works, the first test, run component with no prop should do it, why testing each and single title here? I feel these tests are just fill in lines but not actually increase coverage

simaoalicechen replied:
I am not sure what you meant. I have seen component tests on the previous file before. But if your case is true, then this is the comment that would have been very beneficial on day one

yalexublue replied:
Would you let me know where you have seen such a test is checked in previous test, I'd take a look

yalexublue commented on src/frontend/shortlist/src/components/layout/signup_login/__tests__/LoginSignup.test.js:
```diff
@@ -0,0 +1,317 @@
+  });
+});
+
+describe("Signup.vue", () => {
```
In the future I would suggest separate test for each components, here we have both sign up and log in in this test, but since we are mounting one at a time, there really isn't a reason to combine two tests for two components into one file

simaoalicechen replied:
Sure, I will do that in future

yalexublue commented on src/frontend/shortlist/src/components/layout/signup_login/__tests__/LoginSignup.test.js:
```diff
@@ -0,0 +1,317 @@
+    });
+    const signupForm = componentWrapper.find("#signupForm");
+    expect(signupForm.exists()).toBe(true);
+    await signupForm.trigger("submit.prevent");
```
Same issue, so button exist, you press the button, then ... ?

simaoalicechen replied:
See above

yalexublue commented on src/frontend/shortlist/src/components/layout/signup_login/__tests__/LoginSignup.test.js:
```diff
@@ -0,0 +1,317 @@
+    });
+    const switchToLoginButton = componentWrapper.find("#switchToLoginButton");
+    expect(switchToLoginButton.exists()).toBe(true);
+    await switchToLoginButton.trigger("click");
```
Same reasoning as above

simaoalicechen replied:
See above

yalexublue commented on src/frontend/shortlist/src/components/layout/signup_login/__tests__/LoginSignup.test.js:
```diff
@@ -0,0 +1,317 @@
+    });
+    const signupWithGoogle = componentWrapper.find("#signupWithGoogle");
+    expect(signupWithGoogle.exists()).toBe(true);
+    await signupWithGoogle.trigger("submit.prevent");
```
Add what to expect after trigger please

simaoalicechen replied:
What would you do in this case, as you have experience successfully writing tests? These are the exact guidance I needed.

yalexublue commented on src/frontend/shortlist/src/components/layout/signup_login/__tests__/LoginSignup.test.js:
```diff
@@ -0,0 +1,317 @@
+    await signupWithGoogle.trigger("submit.prevent");
+  });
+
+  it("Check if the big titles work", async () => {
```
Like mentioned before, you don't need to go check if each things in side component exist, if we need to that line by line would take forever, long as the wrapper works fine with and without props, it should be fine imo

simaoalicechen replied:
This was never clearly communicated with me. I feel the button trigger testing would still be beneficial, if you would like to show me how to test it properly in this pr

yalexublue commented on src/frontend/shortlist/src/components/layout/signup_login/__tests__/LoginSignup.test.js:
```diff
@@ -0,0 +1,317 @@
+      },
+    });
+    const stopPropagation = componentWrapper.find("#stopPropagation");
+    expect(stopPropagation.exists()).toBe(true);
```
As mentioned above there are too many existance check for components here, would you also include Daniel in review so we could take a look and see if these check would increase coverage?

simaoalicechen replied:
If you think he can explain better, please feel free to add him

yalexublue reviewed (changes requested):
Detail comments are in the code

simaoalicechen commented:
Will fix the trigger function test and then request additional review and then merge

simaoalicechen committed:
Everything is passing, but without API call, it is kind of lacking.

simaoalicechen committed:
Merge branch 'gcivil-nyu-org:develop' into develop

simaoalicechen committed:
added a file

simaoalicechen committed:
Merge branch 'gcivil-nyu-org:develop' into develop

simaoalicechen committed:
merged

simaoalicechen committed:
2 working tests, one is login, the other one is emit with data'

simaoalicechen committed:
Deleted all commented code. Note the emit test with out data is not working

Next step is to figure out Vuelidate tests

simaoalicechen committed:
Added tests for Signup. appAccountSignup is undefined

simaoalicechen commented:
the emit "appAccountSignup" is undefined and I can't get the firstname, lastname, email, password from it. Is this expected or should we change the codes somehow? @waltage

waltage commented:
> the emit "appAccountSignup" is undefined and I can't get the firstname, lastname, email, password from it. Is this expected or should we change the codes somehow? @waltage

I don't see that being tested in this most recent commit.

Can we remove all the comments so we can see what we've got so far?

simaoalicechen commented:
> > the emit "appAccountSignup" is undefined and I can't get the firstname, lastname, email, password from it. Is this expected or should we change the codes somehow? @waltage
> 
> I don't see that being tested in this most recent commit.
> 
> Can we remove all the comments so we can see what we've got so far?

Do you see the undefined appAccountSignup now?

simaoalicechen committed:
Deleted all comments and added a test file for loggedout

simaoalicechen committed:
ModalFull Screen emit test

Corrected errors in previous files, in the emitted event is undefined, so I wrote a test to say  it is undefined... Does this make sense?

simaoalicechen commented:
Did you see the updates? I guess it might have something to do with the fact that this pr was closed. I reopened it for this purpose. My question is signup emitted event is an empty set. I also tested several other emitted event in other files. Since they do not have data returned to begin with, and they also returned undefined. I just tested toBeTruthy() and toBeUndefined().  @waltage

simaoalicechen committed:
linter fixed

simaoalicechen committed:
added a new test targeting Vuelidate
